### PR TITLE
Fix some cargo bounties accepting/rejecting the wrong items.

### PR DESCRIPTION
**Pull request by Yenwodyah**
## About The Pull Request
* Bounties that were supposed to refuse subtypes of the requested items used to accept them anyways. Now they refuse them correctly.
* Bounties that were supposed to accept subtypes of items that were excluded from acceptance used to still refuse those subtypes. Now they correctly accept them.

## Why It's Good For The Game
Bugfixes

## Changelog
:cl:
fix: A few bounties that accepted/refused the wrong items will now accept/refuse the right ones.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Thoughts, @ArcaneMusic ?

**Comment by ArcaneMusic** [comment]
This appears more correct than my version.

File: code/modules/cargo/bounties/item.dm [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
+	///Individual types that should be accepted even if their supertypes are excluded (yes, apparently this is necessary)
+	var/list/special_include_types
 
 /datum/bounty/item/New()
 	..()
-	wanted_types = typecacheof(wanted_types)
-	exclude_types = typecacheof(exclude_types)
+	wanted_types = typecacheof(wanted_types, only_root_path = !include_subtypes)
+	if (exclude_types)
+		exclude_types = typecacheof(exclude_types)
```
Comment (Rohesie) [inline code/modules/cargo/bounties/item.dm:19]: Bounty datums are not singletons, are they?
```suggestion
		exclude_types = string_assoc_list(typecacheof(exclude_types))
```
If the list doesn't get modified this should provide a nice cache to diminish duplicates.

File: code/modules/cargo/bounties/item.dm [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
+		for (var/e_type in exclude_types)
+			wanted_types[e_type] = FALSE
+	if (special_include_types)
+		for (var/i_type in special_include_types)
+			wanted_types[i_type] = TRUE
+
```
Comment (Rohesie) [inline code/modules/cargo/bounties/item.dm:25]:
```suggestion
			wanted_types[i_type] = TRUE
	wanted_types = string_assoc_list(wanted_types)

```
Same here.

**Comment by Rohesie** [comment]
Make sure `exclude_types` never get modified after being turned into a string list, same for `wanted_types`.
If they do, handle it appropriately, creating new stringlists without modifying the original, as they will be now cached and shared.

Once that is done, I'd say this is ready for merge if it's tested locally and working properly.

**Comment by Yenwodyah** [comment]
Yeah, both of them only get modified in New()

**Pull request merged**